canonn/materialReport.py

```diff
@@ -0,0 +1,75 @@
+# -*- coding: utf-8 -*-
+import threading
+import requests
+from urllib import quote_plus
+import sys
+import json
+from emitter import Emitter
+from debug import Debug
+from debug import debug,error
+
+'''
+{      
+  "system": "string",
+  "body": "string",
+  "latitude": 0,
+  "longitude": 0,
+  "category": "string",
+  "journalName": "string",
+  "journalLocalised": "string",
+  "count": 0,
+  "distanceFromMainStar": 0,
+  "playMode": "string",
+  "isBeta": false,
+  "clientVersion": "string"
+}
+{ "timestamp":"2019-05-16T10:59:15Z", 
+"event":"MaterialCollected", 
+"Category":"Encoded", 
+"Name":"compactemissionsdata", 
+"Name_Localised":"Аномальные компактные данные об излучении", 
+"Count":3 }
+'''
+
+# experimental
+# submitting to a google cloud function
+
+                        
+              
+
+class MeterialsCollected(Emitter):
+    
+    def __init__(self,cmdr, is_beta, system, station, entry,client,lat,lon,body,state):
+        Emitter.__init__(self,cmdr, is_beta, system, None,None,None, entry, body, lat, lon,client)
+        self.modelreport="materialreports"
+        
+    def setPayload(self):
+        payload={}
+        payload["system"]=self.system
+        payload["body"]=  self.body
+        payload["latitude"]=  self.lat
+        payload["longitude"]=  self.lon
+        payload["category"]=self.entry["Category"]
+        payload["journalName"]=self.entry["Name"]
+        #payload["journalLocalised"]=unicode(self.entry.get(u"Name_Localised"))
+        payload["count"]=self.entry["Count"]
+        #payload["distanceFromMainStar"] = #TODO find method to calculate distance
```

Comment:
only should apply to materials on bodies

Reply:
This row disabled